### ISSUE TITLE
LossFunction

### ISSUE THREAD

**Issue by dmccloskey** [dmccloskey/EvoNet#21]
### Description
Abstract base and inherited classes for various loss functions

### Objectives
- [x] cross entropy
- [x] eudlidean distance (autoencoder reconstruction error)
- [x] negative log likelihood
- [x] modified L2
- [x] mean squared error (MSE)
- [x] mean squared logarithmic error (MSLE)
- [x] Kullback Leibler (KL) Divergence
- [ ] BCE with Logits
- [ ] Softmax + CrossEntropy
